lib/dialects/mssql/query-generator.js

```diff
@@ -409,8 +411,15 @@ class MSSQLQueryGenerator extends AbstractQueryGenerator {
     const updateKeys = Object.keys(updateValues);
     const insertKeys = Object.keys(insertValues);
     const insertKeysQuoted = insertKeys.map(key => this.quoteIdentifier(key)).join(', ');
-    const insertValuesEscaped = insertKeys.map(key => this.escape(insertValues[key])).join(', ');
-    const sourceTableQuery = `VALUES(${insertValuesEscaped})`; //Virtual Table
+    const insertValuesEscaped = new CompositionGroup(...insertKeys.map(key => {
```

Comment:
Maybe you should allow passing in an array rather than spreading it all the time.